lib/link.js

```diff
@@ -121,8 +121,9 @@ export default class Link extends Component {
       onClick: this.linkClicked
     }
 
-    // If child is an <a> tag and doesn't have a href attribute we specify it so that repetition is not needed by the user
-    if (child.type === 'a' && !('href' in child.props)) {
+    // If child is an <a> tag (or an <a> styled-component) and doesn't have a href attribute we specify it so that repetition is not needed by the user
+    if ((child.type === 'a' || (child.type === 'function' && child.type.target === 'a')) &&
```

Comment:
I don't like doing any special handling for `styled-component`. I know `styled-component` is pretty good. But this is kind a hack and they could change the `target` field into something else.

And styled-component is just a one solution in this space.

---

But we need to add support for this.

So, I suggest to do it like this:

```js
<Link href="/about" passHref>
  <StyledLink />
</Link>
```